test/test-tuntap.js

```diff
@@ -1,150 +1,38 @@
-#!/usr/bin/env node
+import { TunTap } from "../lib/index.js";
 
-import { TunTap } from '../lib/index.js';
-import { log } from '../lib/logger.js';
+let tun;
+let shuttingDown = false;
 
-// Enable debug logging
-process.argv.push('--debug');
+function cleanupAndExit() {
+  if (shuttingDown) return;
+  shuttingDown = true;
+  try {
+    if (tun && tun.isOpen && !tun.isClosed) {
+      tun.close();
+    }
+  } catch (e) {
+    // Ignore errors during cleanup
+  }
+  process.exit(0);
+}
+
+process.on("SIGINT", cleanupAndExit);
```

Comment:
consider using process.on('exit, ...) callback. It is always executed on exit no matter the signal is. make sure the callback is synchronous though

Reply:
Sure, updated with commmit https://github.com/appium/appium-ios-tuntap/pull/1/commits/c4616b912841f07ca4cce23afad54e996759fc2f